### ISSUE TITLE
kvclient: regression in BenchmarkColBatchScan due to dist sender metrics

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#111142]
I bisected most of the regression on `BenchmarkColBatchScan` from `pkg/sql/colflow` between 23.1 and master to 3502ca2ef2dffd9f19c3bd7d7277324433ac43ad
```
name                        old time/op    new time/op    delta
ColBatchScan/rows=16-24       72.4µs ± 7%    96.4µs ± 5%  +33.18%  (p=0.000 n=10+10)
ColBatchScan/rows=256-24       168µs ± 4%     195µs ± 6%  +15.65%  (p=0.000 n=9+9)
ColBatchScan/rows=4096-24     1.40ms ± 1%    1.41ms ± 1%   +1.23%  (p=0.001 n=8+8)
ColBatchScan/rows=65536-24    21.8ms ± 1%    21.7ms ± 1%     ~     (p=0.481 n=10+10)

name                        old speed      new speed      delta
ColBatchScan/rows=16-24     3.54MB/s ± 7%  2.66MB/s ± 5%  -24.96%  (p=0.000 n=10+10)
ColBatchScan/rows=256-24    24.3MB/s ± 4%  21.0MB/s ± 5%  -13.52%  (p=0.000 n=9+9)
ColBatchScan/rows=4096-24   46.9MB/s ± 1%  46.3MB/s ± 1%   -1.21%  (p=0.001 n=8+8)
ColBatchScan/rows=65536-24  48.1MB/s ± 1%  48.2MB/s ± 1%     ~     (p=0.469 n=10+10)

name                        old alloc/op   new alloc/op   delta
ColBatchScan/rows=16-24       8.21kB ±11%    8.78kB ± 9%     ~     (p=0.053 n=10+9)
ColBatchScan/rows=256-24      28.0kB ± 1%    28.8kB ± 1%   +3.15%  (p=0.000 n=9+9)
ColBatchScan/rows=4096-24      195kB ±14%     196kB ±14%     ~     (p=0.195 n=8+8)
ColBatchScan/rows=65536-24    2.19MB ± 1%    2.19MB ± 1%     ~     (p=0.815 n=9+8)

name                        old allocs/op  new allocs/op  delta
ColBatchScan/rows=16-24          114 ±10%       119 ± 6%     ~     (p=0.076 n=10+9)
ColBatchScan/rows=256-24         158 ± 1%       168 ± 1%   +6.55%  (p=0.000 n=9+9)
ColBatchScan/rows=4096-24       246 ±130%      254 ±127%   +3.61%  (p=0.046 n=8+8)
ColBatchScan/rows=65536-24       292 ±24%       326 ±39%  +11.82%  (p=0.035 n=8+8)
```
My guess (based on a quick glance at the code) is that this is due to having to compute `ba.Size()` and `br.Size()`, so we might not be able to do much about it, but will defer to KV.

Jira issue: CRDB-31787

**Comment by yuzefovich** [comment]
This regression can also be observed on `BenchmarkKV/Scan/Native` in `pkg/sql/tests`:
```
name                          old time/op    new time/op    delta
KV/Scan/Native/rows=1-24        29.3µs ± 1%    43.6µs ± 1%  +48.61%  (p=0.000 n=10+9)
KV/Scan/Native/rows=10-24       32.1µs ± 2%    46.5µs ± 2%  +44.65%  (p=0.000 n=10+10)
KV/Scan/Native/rows=100-24      50.1µs ± 1%    66.3µs ± 2%  +32.34%  (p=0.000 n=9+10)
KV/Scan/Native/rows=1000-24      227µs ± 3%     252µs ± 1%  +10.88%  (p=0.000 n=10+9)
KV/Scan/Native/rows=10000-24    1.91ms ± 4%    2.02ms ± 2%   +5.73%  (p=0.000 n=9+9)

name                          old alloc/op   new alloc/op   delta
KV/Scan/Native/rows=1-24        8.35kB ± 0%    9.33kB ± 0%  +11.72%  (p=0.000 n=10+10)
KV/Scan/Native/rows=10-24       9.79kB ± 0%   10.77kB ± 0%   +9.96%  (p=0.000 n=10+9)
KV/Scan/Native/rows=100-24      22.6kB ± 0%    23.7kB ± 0%   +4.79%  (p=0.000 n=8+10)
KV/Scan/Native/rows=1000-24      176kB ± 0%     177kB ± 0%   +0.71%  (p=0.000 n=9+10)
KV/Scan/Native/rows=10000-24    1.55MB ± 0%    1.55MB ± 0%   +0.11%  (p=0.000 n=10+10)

name                          old allocs/op  new allocs/op  delta
KV/Scan/Native/rows=1-24          61.0 ± 0%      74.0 ± 0%  +21.31%  (p=0.000 n=10+8)
KV/Scan/Native/rows=10-24         66.0 ± 0%      79.0 ± 0%  +19.70%  (p=0.000 n=8+8)
KV/Scan/Native/rows=100-24        74.1 ± 1%      88.3 ± 1%  +19.12%  (p=0.000 n=8+10)
KV/Scan/Native/rows=1000-24        126 ± 4%       143 ± 6%  +13.81%  (p=0.000 n=9+10)
KV/Scan/Native/rows=10000-24       597 ± 3%       622 ± 2%   +4.09%  (p=0.000 n=10+10)
```

**Comment by kvoli** [comment]
Will be closed on #113302 merging.

**Comment by kvoli** [comment]
Closed on https://github.com/cockroachdb/cockroach/pull/113302.
